controllers/managedocs_controller.go

```diff
@@ -146,3 +156,31 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) updateComponents(req ctrl.Request) (v1.ComponentStatusMap, error) {
+	var components v1.ComponentStatusMap
+
+	// Checking the StorageCluster component.
+	var sc ocsv1.StorageCluster
+
+	scNamespacedName := types.NamespacedName{
+		Name:      storageClusterName,
+		Namespace: req.Namespace,
+	}
+
+	err := r.Get(r.ctx, scNamespacedName, &sc)
+	if err == nil {
+		if sc.Status.Phase == "Ready" {
+			components.StorageCluster.Status = v1.ComponentReady
+		} else {
+			components.StorageCluster.Status = v1.ComponentPending
+		}
+	} else if errors.IsNotFound(err) {
+		components.StorageCluster.Status = v1.ComponentNone
```

Comment:
Let's be more specific, can we set the state to `NotFound` as everything else is treated as `Unknown`?